### ISSUE TITLE
TabBarStyle zIndex property doesn't have any effect

### ISSUE THREAD

**Issue by romancxx** [react-community/react-navigation#10064]
### Current behavior

I am trying to put a `LinearGradient` behind my bottom tab to make it fade over the screen. I recently upgraded to v6 and realized that this code is no longer working. To achieve that, I am using `zIndex` in the `tabBarStyle` of the `screenOptions` with a higher value than my linear. But whatever the value I pass, it doesn't change anything as it used to in v5. Now the linear is either on top of the bottom tab or under everything so not visible at all.

 
![Screenshot 2021-10-19 at 19 20 46](https://user-images.githubusercontent.com/21194324/137960421-577d6dcb-ef4c-4bdb-bef3-13fdafb69c94.png)

  

### Expected behavior

It used to look like that with v5 : 

![Screenshot 2021-10-19 at 19 17 42](https://user-images.githubusercontent.com/21194324/137960002-c07b2b31-bf68-4a19-9276-5d037e751ce6.png)

Here is a link reproducing the behavior I had in v5 : https://snack.expo.dev/@flowwwww/zindex-react-nav-v5

Maybe I'm trying to achieve this the wrong way but couldn't find any good example to put a view behind the bottom tab. 

### Reproduction

https://snack.expo.dev/@flowwwww/8ea5ba

### Platform

- [X] Android
- [X] iOS
- [X] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [X] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-bottom-tabs
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |  ^6.0.6
| @react-navigation/bottom-tabs          | ^6.0.9
| @react-navigation/stack                |^6.0.11
| @react-navigation/native-stack         | 6.2.5
| react-native-safe-area-context         |3.2.0
| react-native-screens                   |^3.8.0
| react-native                           | 0.64.1

**Comment by Nazort** [comment]
Did you manage to find a fix @Tatsiana0607 ? 

The only solution I personally see would be to do a custom tab nav component instead of using the one provided by react-navigation, which is sad since it is just to fix a small detail ...

**Comment by Tatsiana0607** [comment]
@Nazort No, unfortunately I didn't.
I decided to ignore this issue for now cause like you said it requires a lot of changes from my side to fix such a little bug. So I just hope that contributors will find a way to fix it someday...

**Comment by react-one** [comment]
The issue is due to a change in this PR [ [react-navigation#9793](https://github.com/react-navigation/react-navigation/pull/9793) ] [line 42](https://github.com/aleppos/react-navigation/blob/12e4f2ded9e24c3cbff6dca116a54620cfebcd8a/packages/elements/src/SafeAreaProviderCompat.tsx#L42). Where the children were wrapped in a View Component. The zIndex behavior will not work since the children elements are not at the same level. I created a tested PR fix that handles that. And it needs to be discussed.

## The right way to do it
Otherwise, The same wanted implementations for the shadow effect can be implemented easily using tabBar custom render prop. You import <BottomTabBar /> and add anything else to it. That can be the right way to do such things. And not just a workaround. Here is a clear example in V6:

```tsx
// …
import {
  createBottomTabNavigator,
  BottomTabBar,  // <------- import the native component
} from '@react-navigation/bottom-tabs';
// …

function MyTabs() {
  return (
    <>
      <Tab.Navigator
        tabBar={(props) => {
          return (
            // ------------------------------------------
            <>
              <LinearGradient
                colors={['transparent', 'red', 'red']}
                style={{
                  position: 'absolute',
                  bottom: 0,
                  left: 0,
                  right: 0,
                  height: 75,
                  zIndex: 1,
                }}
              />
              <BottomTabBar {...props} />
            </>
            // ------------------------------------------
          );
        }}
        screenOptions={{
          tabBarStyle: {
            backgroundColor: 'black',
            borderTopColor: 'transparent',

            borderTopLeftRadius: 20,
            borderTopRightRadius: 20,
            position: 'absolute',
            height: 60,
            marginTop: 20,
            overflow: 'visible',
            paddingHorizontal: 15,
            zIndex: 2,
          },
          tabBarItemStyle: {
            paddingVertical: 10,
          },
          tabBarLabelStyle: {
            fontSize: 11,
          },
        }}
      >
        <Tab.Screen
          name="Home"
          component={() => {
            return (
              <View style={{ flex: 1 }}>
                <Text style={styles.paragraph}>Home</Text>
              </View>
            );
          }}
        />
        <Tab.Screen
          name="Settings"
          component={() => {
            return (
              <View style={{ flex: 1 }}>
                <Text style={styles.paragraph}>Settings</Text>
              </View>
            );
          }}
        />
      </Tab.Navigator>
      {/* </View> */}
    </>
  );
}
```

Basically, we are just extending the default one [packages/bottom-tabs/src/views/BottomTabView.tsx#L36](https://github.com/aleppos/react-navigation/blob/12e4f2ded9e24c3cbff6dca116a54620cfebcd8a/packages/bottom-tabs/src/views/BottomTabView.tsx#L36).

```tsx
export default function BottomTabView(props: Props) {
  const {
    tabBar = (props: BottomTabBarProps) => <BottomTabBar {...props} />,
    state,
```

props will be received the same way.

**Comment by Tatsiana0607** [comment]
@react-one Both ways work perfectly for me. Thanks for your help!
